variables.tf

```diff
@@ -2,6 +2,16 @@
 # The above line disables rule "Name Is Not Snake Case" in KICS
 
 ### required Variables
+variable "deployment_type" {
+  type        = string
+  description = "Define the deployment type of either EC2 or ECS"
+
+  validation {
+    condition     = contains(["ec2", "ecs"], var.deployment_type)
+    error_message = "Invalid value - please choose ec2 or ecs."
+  }
+}
+
 variable "ami_id" {
   description = "AMI image id to use for the deployments"
   type        = string
```

Comment:
EC2 specific variables will need to become option at this level and required at the submodule level

Reply:
EC2 specific variables now optional